src/Strategy/Googlesource/RepoManifest.hs

```diff
@@ -0,0 +1,218 @@
+{-# language TemplateHaskell #-}
+
+module Strategy.Googlesource.RepoManifest
+  ( discover
+  , buildGraph
+  , analyze
+  , validatedProject
+  , validatedProjects
+  , nestedValidatedProjects
+
+  , RepoManifest(..)
+  , ManifestRemote(..)
+  , ManifestDefault(..)
+  , ManifestProject(..)
+  , ValidatedProject(..)
+
+  , mkProjectClosure
+  ) where
+
+import Prologue
+
+import Control.Carrier.Error.Either
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+
+import DepTypes
+import Discovery.Walk
+import Effect.ReadFS
+import Graphing (Graphing, unfold)
+import Parse.XML
+import Types
+
+-- We're looking for a file called "manifest.xml" in a directory called ".repo"
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \_ subdirs files -> do
+  case find (\f -> (==) "manifest.xml" (fileName f)) files of
+    Nothing -> walkContinue
+    Just file -> do
+      if (dirname $ parent file) == $(mkRelDir ".repo") then do
+        runSimpleStrategy "googlesource-repomanifest" GooglesourceGroup $ analyze file
+        walkSkipAll subdirs
+      else walkContinue
+
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m, MonadFail m) => Path Rel File -> m ProjectClosureBody
+analyze file = do
```

Comment:
[style] another way to write this is
```hs
analyze file = mkProjectClosure file <$> nestedValidatedProjects file
```

Anytime you see:
```hs
something = do
  bar <- doSomething
  pure (foo bar)
```
it's the same as
```hs
something = foo <$> doSomething
```

Reply:
Nice!